modules/dxp/apps/commerce/commerce-shop-by-diagram/commerce-shop-by-diagram-web/src/main/resources/META-INF/resources/js/DiagramFooter.js

```diff
@@ -12,26 +12,43 @@
 import ClayButton from '@clayui/button';
 import {ClaySelect} from '@clayui/form';
 import ClayIcon from '@clayui/icon';
-import React from 'react';
+import React, {useContext} from 'react';
+
+import DiagramCommonContext from './context/DiagramCommonContext';
 
 const OPTIONS = [2, 1.75, 1.5, 1.25, 1, 0.75, 0.5];
 
-const DiagramFooter = ({
-	enableResetZoom,
-	selectedOption,
-	setChangedScale,
-	setResetZoom,
-	setSelectedOption,
-	setZoomInHandler,
-	setZoomOutHandler,
-}) => {
+const DiagramFooter = () => {
 	const handleZoomChange = (event) => {
 		setSelectedOption(event.target.value);
 		setChangedScale(true);
 	};
 
+	const {
+		enableResetZoom,
+		isAdmin,
+		selectedOption,
+		setAddPinHandler,
+		setChangedScale,
+		setResetZoom,
+		setSelectedOption,
+		setZoomInHandler,
+		setZoomOutHandler,
+	} = useContext(DiagramCommonContext);
+
 	return (
 		<div className="d-flex diagram-footer justify-content-end mt-3">
+			{isAdmin && (
+				<ClayButton
+					className="mr-3"
+					onClick={() => setAddPinHandler(true)}
+				>
+					<span className="inline-item inline-item-before">
+						<ClayIcon symbol="pin" />
+					</span>
+					{Liferay.Language.get('add-pin')}
```

Comment:
Empty line between siblings

Reply:
sf didn't add it automatically

Reply:
Doesn't matter, still need to have it. A lot of things aren't automatic but we still need to do them